docs/web-apps/automated-testing/cucumberjs-playwright/yaml.md

```diff
@@ -578,6 +578,35 @@ Do not use `dependencies` and `packages` at the same time.
 
 ---
 
+### `usePackageLock`
+
+<p><small>| OPTIONAL | BOOLEAN |</small></p>
+
+Specifies whether to use the project's package-lock.json when installing npm
+dependencies. If true, package-lock.json will be used during package
+installation which can improve the speed of installation.
+
+To use this feature, additional pre-requisites must be met:
+* A package-lock.json must be present in your project.
+* The `@playwright/test` version in your package.json must **exactly** match
+the version defined in your saucectl config.
+
+```yaml
+npm:
+  usePackageLock: true
+  packages:
+    lodash: "4.17.20"
+    "@babel/preset-typescript": "7.12"
+```
+
+:::note
+You can optionally omit the `packages` configuration when using
+`usePackageLock`. If omitted, the packages defined in your `package.json` will
+be installed.
```

Comment:
I wonder if it's better to flip the wording on this one? This configuration section is specifically about `usePackageLock`, with the use of `packages` being optional. Rather than optionally omitting the `packages` setting, it's optionally available to add additional packages that are not declared in the `package-lock.json`.

This ought to be discouraged.

Ideally, you wouldn't want to use `packages`, since those additions happen at runtime and in our testing, anything that isn't part of the `package-lock.json` takes much longer to install due to NPM having to modify the dependency tree. Users should really migrate over to `usePackageLock` sans `packages` if they can.